### PR TITLE
DFR-969: Add data to ccd for solicitor contact details update

### PR DESCRIPTION
**Pull request by FaisalMoJ**
Add data to ccd for updating solicitor contact details. Also adds a
new object in CaseFields to have a user role. Possible two values
for solicitor contact details would be [APPSOLICITOR] or
[RESPSOLICITOR] based on whether solicitor is applicant or respondent

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DFR-969

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: definitions/consented/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json [hunk 1]
```diff
@@ -1370,5 +1370,12 @@
     "CaseEventID": "FR_TransferredToLocalCourt",
     "UserRole": "caseworker-divorce-financialremedy-judiciary",
     "CRUD": "R"
+  },
```
Comment (gmikerlemos) [inline definitions/consented/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json:1373]: this should be behind a feature flag, right?

File: definitions/consented/json/AuthorisationCaseField/AuthorisationCaseField.json [hunk 2]
```diff
@@ -5612,5 +5612,47 @@
     "CaseFieldID": "civilPartnership",
     "UserRole": "caseworker-caa",
     "CRUD": "CRU"
+  },
```
Comment (gmikerlemos) [inline definitions/consented/json/AuthorisationCaseField/AuthorisationCaseField.json:5615]: Probs these should be behind a FF as well...

File: definitions/contested/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json [hunk 3]
```diff
@@ -1393,6 +1393,34 @@
     "UserRole": "caseworker-divorce-financialremedy-solicitor",
     "CRUD": "R"
   },
+  {
```
Comment (gmikerlemos) [inline definitions/contested/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json:1396]: behind a FF?

**Review by gmikerlemos — CHANGES_REQUESTED** [review]
You need to put these changes behind a feature flag

File: definitions/consented/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json (hunk 1 above)
Comment (FaisalMoJ) [reply definitions/consented/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json]: Moved it behind feature flag as suggested

**Review by gmikerlemos — APPROVED** [review]

**Comment by FaisalMoJ** [comment]
Closing this PR as this ticket is now tracked by #740

**Pull request closed without merging**